diff --git a/node_modules/history/cjs/history.js b/node_modules/history/cjs/history.js
index fcd8eba..4df20d5 100644
--- a/node_modules/history/cjs/history.js
+++ b/node_modules/history/cjs/history.js
@@ -103,16 +103,6 @@ function createLocation(path, state, key, currentLocation) {
     if (state !== undefined && location.state === undefined) location.state = state;
   }
 
-  try {
-    location.pathname = decodeURI(location.pathname);
-  } catch (e) {
-    if (e instanceof URIError) {
-      throw new URIError('Pathname "' + location.pathname + '" could not be decoded. ' + 'This is likely caused by an invalid percent-encoding.');
-    } else {
-      throw e;
-    }
-  }
-
   if (key) location.key = key;
 
   if (currentLocation) {
diff --git a/node_modules/history/esm/history.js b/node_modules/history/esm/history.js
index df67820..e0e0d4f 100644
--- a/node_modules/history/esm/history.js
+++ b/node_modules/history/esm/history.js
@@ -80,16 +80,6 @@ function createLocation(path, state, key, currentLocation) {
     if (state !== undefined && location.state === undefined) location.state = state;
   }
 
-  try {
-    location.pathname = decodeURI(location.pathname);
-  } catch (e) {
-    if (e instanceof URIError) {
-      throw new URIError('Pathname "' + location.pathname + '" could not be decoded. ' + 'This is likely caused by an invalid percent-encoding.');
-    } else {
-      throw e;
-    }
-  }
-
   if (key) location.key = key;
 
   if (currentLocation) {
